examples/4_simple_storage.py

```diff
@@ -0,0 +1,157 @@
+            event.type == EventType.CHAT
+            and (members_type == "team" or members_type == "impteamnative")
+        ):
+            return
+
+        channel = event.msg.channel
+
+        # support teams and implicit self teams
+        team = (
+            channel.name if members_type == "team" else "{0},{0}".format(channel.name)
+        )
+
+        msg = event.msg.content.text.body.split(" ")
+        if msg[0] != self.MSG_PREFIX:
+            return
+
+        if len(msg) == 2:
+            if msg[1] == KVMsg.LIST.value:
+                # !storage list
+                res = await bot.kvstore.list_namespaces(team)
+                await bot.chat.send(channel, str(res))
+                return
+            if msg[1] == KVMsg.HELP.value:
+                # !storage help
+                send_msg = "Available commands:\
+                    \n`!storage put <namespace> <key> <value> (<revision>)`\
+                    \n`!storage get <namespace> <key>`\
+                    \n`!storage delete <namespace> <key> (<revision>)`\
+                    \n`!storage list`\
+                    \n`!storage list <namespace>`"
+                await bot.chat.send(channel, send_msg)
+                return
+        if len(msg) == 3 and msg[1] == KVMsg.LIST.value:
+            # !storage list <namespace>
+            namespace = msg[2]
+            res = await bot.kvstore.list_entrykeys(team, namespace)
+            await bot.chat.send(channel, str(res))
+            return
+        if len(msg) == 4:
+            (action, namespace, key) = (msg[1], msg[2], msg[3])
+            if action == KVMsg.GET.value:
+                # !storage get <namespace> <key>
+                res = await bot.kvstore.get(team, namespace, key)
+                await bot.chat.send(channel, str(res))
+                return
+        if len(msg) == 4 or len(msg) == 5:
+            (action, namespace, key) = (msg[1], msg[2], msg[3])
+            revision = int(msg[4]) if len(msg) == 5 else None
+            if action == KVMsg.DELETE.value:
+                # !storage delete <namespace> <key> (<revision>)
+                try:
+                    res = await bot.kvstore.delete(
+                        team, namespace, key, revision=revision
+                    )
+                    await bot.chat.send(channel, str(res))
+                except Exception as e:
+                    await bot.chat.send(channel, str(e))
+                return
+        if len(msg) == 5 or len(msg) == 6:
+            # !storage put <namespace> <key> <value> (<revision>)
+            (action, namespace, key, value) = (msg[1], msg[2], msg[3], msg[4])
+            revision = int(msg[5]) if len(msg) == 6 else None
+            if action == KVMsg.PUT.value:
+                try:
+                    # note: if revision=None, the server does a get (to get
+                    # the latest revision number) then a put (with revision
+                    # number + 1). this operation is not atomic.
```

Comment:
i think this might be more information than people need for the vast majority of usecases, and it may serve to confuse more than inform. i wouldn't go to the mat over this, but slight preference for completely leaving out revision from here. if people want concurrency, they'll be looking at the more advanced examples anyway.

Reply:
i see what you mean, but i included it here because it actually helps a lot with debugging (when writing the other bot examples, i'd also have this bot running to check my kv store state easily). plus this was a notable point to me, so that's why i decided to write a comment about it.